helix-core/src/main/java/org/apache/helix/manager/zk/client/HelixZkClient.java

```diff
@@ -148,6 +189,36 @@ void asyncSetData(final String path, Object datat, final int version,
 
   PathBasedZkSerializer getZkSerializer();
 
+  /**
+   * Creates a {@link org.I0Itec.zkclient.IZkStateListener} that wraps a default implementation
+   * of {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener}, which means the returned
+   * listener runs the methods of {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener}.
+   * This is for backwards compatibility and to avoid breaking the original implementation of
+   * {@link org.I0Itec.zkclient.IZkStateListener}.
+   *
+   * @param listener {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener}
+   * @return {@link org.I0Itec.zkclient.IZkStateListener}
+   */
+  static IZkStateListener createDefaultIZkStateListener(
```

Comment:
As commented previously, I think the intent of this function would be clearer if the return type was expressed as `org.I0Itec.zkclient.IZkStateListener`. Perhaps import our own listener instead?

Reply:
I was considering this. My thought was:
1. If we import the new listener, we have to change `subscribeStateChanges(final IZkStateListener listener)` to `subscribeStateChanges(final org.I0Itec.zkclient.IZkStateListener listener)` as well. I was trying to keep the changes minimal, and make the default method `default void subscribeStateChanges(final org.apache.helix.manager.zk.zookeeper.IZkStateListener listener)` look like a new method with more distinction.
2. If we don't have other I0Itec imports in this interface, I believe it looks cleaner to just import our own listener. So we could completely get rid of I0Itec in this interface.

This is just my thought. We can also involve @jiajunwang @lei-xia to comment.